### DISCUSSION TITLE
Transformation

### DISCUSSION THREAD

**Discussion by john412smith** [iptv-org/community#545] [Show and tell]
Hello, after finding your iptv repository, I had some ideas and started writing some code. I love the app called "IPTV Smarters Lite", in the way it displays and plays content. However this was created for illegal re-stream server called Xtream Codes. Thought to myself what if I could use it for legal and freeview content without using XC's server itself.

Step 1, build XCAPI emulator. 
Step 2, get legal viewable live streams and vod content. 
Step 3, take a break eat some cake and have a tea. 
Step 4, import data, filtering, cataloging and screening. 
Step 5, slam head on wall till can login with Smarters app.

In the end, this is my results so far. Install IPTV Smarters Lite, and login with xtream codes login.

Username: preview
Password: test
url: http://yz.sytes.net:2800

You can also get an m3u at 
http://yz.sytes.net:2800/m3u/preview/test

Live channels from IPTV-ORG/iptv are working, the movies section is working with streams from public sources such as Tubi, Xumo, and others.

The series section... XC uses a horrible data structure making it a massive headache to do. I have mapped the structures, but converting the data to it is proving difficult and I cannot afford to much of my time right now to complete it. The series are Listed, but not playable.

There is much work to do on this project that I am calling "XCFreEMU", soon as I get a working series section, I will be throwing it up here on github so anyone can use.

Hope you enjoy, and please let me know of any issues you find... but keep in mind, code is still very early so might not be functioning some times as I'm updating things.

**Comment by thihazawsatellite** [comment]
thank brother

**Comment by georgoskar** [comment]
#EXTINF:-0 tvg-id="ext" group-title="channels",baraza FHD greek music hits
https://eco.streams.ovh:443/BarazaTV/BarazaTV/playlist.m3u8
#EXTINF:-1 tvg-id="ext" group-title="channels",baraza FHD greek 2
https://eco.streams.ovh/BarazaTV/BarazarazaTV/BarazaTV/playlist.m3u8
#EXTINF:-1 tvg-id="ext" group-title="channels",Baraza Relaxing music
https://media2.streambrothers.com:1936/8106/8106/playlist.m3u8
#EXTINF:-1 tvg-id="ext" group-title="channels",Baraza radio deep house
https://i6.streams.ovh/sc/barazar1/stream
#EXTINF:-1 tvg-id="ext" group-title="channels",jacksonpallace radio
http://jacksonspalace.radioca.st:8139/stream
#EXTINF:-1 tvg-id="ext" group-title="channels",Baraza radio deep house
https://i6.streams.ovh:16074/stream

**Comment by john412smith** [comment]
Sorry everyone, getting ready to move this weekend, out of the country, will try to upload the project by Friday. Due to DDoS, use http://wx.sytes.net:2800 for now, when I get settled will put it up on a real server for everyone to enjoy.

Side note: Movie playback depends on your device DRM capabilities.

**Comment by john412smith** [comment]
Well move was completed, however, the HDD with the files on it died, and the backup drive got lost. So.... I'm restarting the project from scratch, I hope to have it back up soon as possible. My apologies.

**Reply by VictorBuica** [reply]
Hey is there any update on the project. If you need a helping hand I can try move things around if you'd like.

**Reply by john412smith** [reply]
All good, my brother managed to find an exact model drive on ebay (he has a lot of free time...), exact down to month/year manufactured. Going to attempt to repair the control board (physical damaged chip on board) of the disk to recover. 5 weeks to receive the drive ugh.

If that fails, I have already started rebuilding, I still remember the 9133223423 ways it didn't work, so progress is breezing by. By end of month hope to have a new server up and running with at least login and LiveTV section working.

**Reply by milutii** [reply]
Hello my fellow Canadian - any chance you've had time to work on this? :)
